modules/elements-color-picker/assets/js/editor/commands/show-swatches.js

```diff
@@ -0,0 +1,157 @@
+import CommandBase from 'elementor-api/modules/command-base';
+import Utils from '../../../../../../core/app/assets/js/utils/utils';
+
+export class ShowSwatches extends CommandBase {
+	constructor( args ) {
+		super( args );
+
+		this.colors = {};
+		this.pickerClass = 'elementor-element-color-picker';
+		this.pickerSelector = '.' + this.pickerClass;
+		this.container = null;
+		this.tmpImages = [];
+
+		this.COLORS_LIMIT = 5;
+	}
+
+	apply( args ) {
+		if ( ! args.id ) {
+			return;
+		}
+
+		this.container = elementor.getContainer( args.id );
+
+		if ( this.container.view.$el.find( this.pickerSelector ).length ) {
+			return;
+		}
+
+		this.extractColorsFromSettings();
+
+		setTimeout( () => {
+			this.extractColorsFromImages();
+			this.initSwatch();
+		}, 100 );
+	}
+
+	extractColorsFromSettings() {
+		// Iterate over the widget controls.
+		Object.keys( this.container.settings.attributes ).map( ( control ) => {
+			// Limit colors count.
+			if ( this.reachedColorsLimit() ) {
+				return;
+			}
+
+			if ( ! ( control in this.container.controls ) ) {
+				return;
+			}
+
+			// Throw non-active controls.
+			if ( ! elementor.helpers.isActiveControl( this.container.controls[ control ], this.container.settings.attributes ) ) {
+				return;
+			}
+
+			// Handle background images.
+			if ( control.startsWith( '_background_image' ) ) {
+				this.addTempBackgroundImage( this.container.getSetting( control ) );
+			}
+
+			// Throw non-color controls.
+			if ( 'color' !== this.container.controls[ control ]?.type ) {
+				return;
+			}
```

Comment:
Move it before the check of `isActiveControl` because these checks are cheaper in performance terms.